### DISCUSSION TITLE
Endpoint pagination with MongoDB (Motor)

### DISCUSSION THREAD

**Discussion by Robin-Bonnin** [fastapi/fastapi#6589] [Questions]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
class ListModel:
    @classmethod
    async def list(
        cls, filters={}, sorting_field="_id", page_nb=0, page_size=10, sorting_type=1
    ):
        return (
            await cls.Meta.collection.find(filter=filters)
            .sort(sorting_field, sorting_type)
            .skip((page_nb - 1) * page_size if page_nb > 0 else 0)
            .limit(page_size)
            .to_list(page_size)
        )



class EmailThreadModel(DBModelMixin,UpdateModel, CreateModel, ListModel, GetModel, DeleteModel):
   ** All the fields I want to store in DB come here **

    class Meta:
        collection = EmailThreadCollection
    # I override a value here specific to this model
    async def create(self):
        self.last_event_at = datetime.now(tz=timezone.utc)
        return await super().create()



# Here I want to return the list of threads
# However I cannot find a good way to paginate my results
@router.get("/", response_model=List[EmailThreadGet])
async def get_list_of_threads():
    thread_list = await EmailThreadModel.list(filters={"client_id": 515})
    print(thread_list)
    if thread_list is None:
        raise HTTPException(status_code=status.HTTP_404_NOT_FOUND)

    return thread_list
```

### Description

Hello,

I am implementing a new app in FastAPI. It will run using MongoDB (via Motor) and I am running into troubles to paginate my list endpoints.

You can see attached an example of code that I have for one of my models. 
I know I can paginate the queries done to Mongo but I have a hard time translating that to my endpoint ensuring consistent and correct results. 

Is there any good package existing for this? 

PS: If there is any good tooling library to work on FastAPI with MongoDB I am listening! 

Thank you!

### Operating System

Linux

### FastAPI Version

0.68.1

### Python Version

3.8.10

**Comment by trevorwang** [comment]
here's a lib that depends on sqlalchemy.    https://github.com/uriyyo/fastapi-pagination

May give you some inspiration.

**Reply by dusktreader** [reply]
Unfortunately, while that library does support Motor to some degree, it doesn't support `AsyncIOMotorCursor` that you get from `collection.find()`.
